etcdserver/server.go

```diff
@@ -1002,6 +1002,26 @@ func (s *EtcdServer) LeaderStats() []byte {
 
 func (s *EtcdServer) StoreStats() []byte { return s.store.JsonStats() }
 
+func (s *EtcdServer) checkMembershipOperationPermission(ctx context.Context) error {
```

Comment:
Huh? Why would all that be necessary? It was working fine allowing `authStore == nil` in the tests. The `server_test.go` membership tests now need `!= nil` plus the dummy auth code because this function does not do a nil check. If there's a nil check in this function, all that new code can be removed, right?